commitlint.config.ts

```diff
@@ -61,6 +62,53 @@ module.exports = {
 
         {
             rules: {
+                'body-prose': ({body}: {body:any}) => {
+                    let offence = false;
+
+                    // does msg have a body?
+                    if (body !== null) {
+                        let bodyStr = convertAnyToString(body, "body");
+                        
+                        for (let par of bodyStr.trim().split('\n\n')){
+
+                            // It's a paragraph that only consists of a block
+                            if (/^```[^]*```$/.test(par.trim())){
+                                continue;
+                            }
+
+                            par = par.replace(/```[^]*```/g, '').trim();
+
+                            let firstIsUpperCase = par[0].toUpperCase() == par[0];
+                            let firstIsLowerCase = par[0].toLowerCase() == par[0];
+                            let startWithLowerCase = firstIsLowerCase && (!firstIsUpperCase);
+
+                            let endsWithDotOrColon = par[par.length - 1] === '.' || par[par.length - 1] === ':';
+                            
+                            if (startWithLowerCase || !endsWithDotOrColon){
+                                let line = par.split(/\r?\n/)[0];
+                                
+                                // it's a URL
+                                let containsASpace = line.indexOf(" ") >= 0;
+                                
+                                // it's a footer reference, i.e. [1] someUrl://foo/bar/baz
+                                let startsWithRef = (line[0] === "[" && line.indexOf("] ") > 0);
```

Comment:
@realmarv I believe you're copying code from other rules here, right? if so, let's not duplicate code, let's create functions that both rules use please

Reply:
Done!